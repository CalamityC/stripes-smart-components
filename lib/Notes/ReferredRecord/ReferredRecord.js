import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { KeyValue } from '@folio/stripes-components';

import styles from './ReferredRecord.css';

import { referredEntityDataShape } from '../NoteForm/noteShapes';

const ReferredRecord = (props) => {
  const {
    referredEntityData: {
      name,
      type,
    },
    entityTypeTranslationKeys,
  } = props;

  const entityType = (
    <span className={styles['referred-record__entity-type']}>
      <FormattedMessage id={entityTypeTranslationKeys[type]} />
    </span>
  );

  return (
    <KeyValue label={entityType}>
      {name}
    </KeyValue>
  );
};

ReferredRecord.propTypes = {
  entityTypeTranslationKeys: PropTypes.objectOf(PropTypes.string),
  referredEntityData: referredEntityDataShape.isRequired,
};

export default ReferredRecord;
