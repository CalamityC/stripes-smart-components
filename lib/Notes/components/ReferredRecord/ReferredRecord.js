import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { KeyValue } from '@folio/stripes-components';

import { referredEntityDataShape } from '../NoteForm/noteShapes';

import styles from './ReferredRecord.css';

const ReferredRecord = (props) => {
  const {
    referredEntityData: {
      name,
      type,
    },
    entityTypeTranslationKeys,
  } = props;

  const entityType = (
    <span data-testid="entityType" className={styles['referred-record__entity-type']}>
      <FormattedMessage id={entityTypeTranslationKeys[type]} />
    </span>
  );

  return (
    <KeyValue label={entityType}>
      <span data-testid="entityName">
        {name}
      </span>
    </KeyValue>
  );
};

ReferredRecord.propTypes = {
  entityTypeTranslationKeys: PropTypes.objectOf(PropTypes.string),
  referredEntityData: referredEntityDataShape.isRequired,
};

export default ReferredRecord;
