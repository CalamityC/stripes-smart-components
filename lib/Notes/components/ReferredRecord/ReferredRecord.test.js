import React from 'react';
import { render } from '@testing-library/react';
import 'jest-dom/extend-expect';
import { IntlProvider } from 'react-intl';
import '@testing-library/react/cleanup-after-each';

import ReferredRecord from './ReferredRecord';

const translations = {
  'package-translation-key': 'Package'
};

const entityTypeTranslationKeys = {
  package: 'package-translation-key',
};

const referredEntityData = {
  id: '123',
  name: 'Some name',
  type: 'package'
};

const renderComponent = () => render(
  <IntlProvider
    locale="en"
    messages={translations}
  >
    <ReferredRecord
      entityTypeTranslationKeys={entityTypeTranslationKeys}
      referredEntityData={referredEntityData}
    />
  </IntlProvider>
);

describe('ReferredRecord', () => {
  let container, getByTestId;

  beforeEach(() => {
    ({ container, getByTestId } = renderComponent());
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render referred entity name', () => {
    const entityNameWrapper = getByTestId('entityName');
    const entityName = entityNameWrapper.textContent;

    expect(entityName).toBe(referredEntityData.name);
  });

  it('should render referred entity type', () => {
    const entityTypeWrapper = getByTestId('entityType').getElementsByTagName('span')[0];
    const entityType = entityTypeWrapper.textContent;

    expect(entityType).toBe(translations[entityTypeTranslationKeys[referredEntityData.type]]);
  });
});
