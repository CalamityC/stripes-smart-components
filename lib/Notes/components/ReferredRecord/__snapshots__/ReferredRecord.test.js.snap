// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferredRecord should match snapshot 1`] = `
<div>
  <div
    class="kvRoot"
  >
    <div
      class="kvLabel"
    >
      <span
        class="referred-record__entity-type"
        data-testid="entityType"
      >
        <span>
          Package
        </span>
      </span>
    </div>
    <span
      data-testid="entityName"
    >
      Some name
    </span>
  </div>
</div>
`;
